### PR TITLE
Redesign search

### PR DESCRIPTION
**Pull request by tysongach**
- Moved the search into the main content header to make better use of
  vertical space in the layout.
- Search is now visually (and in the DOM) _after_ the page title, which 
  makes for a more logical and expected flow because it is a page-based
  search.
- Improved accessibility:
  - Navigating the search via keyboard now works in Safari.
  - Meaningful accessible names and labels were added.
  - Include the name of the resource being searched in labels.
- Reduced the CSS needed for search by 44%.
- Remove unnecessary `content_for` helper that wrapped the rendering of
  the search partial.

**Note:** This PR depends on #880 being merged.

## Before

screen shot 2017-05-10 at 16 12 30

## After

screen shot 2017-05-10 at 16 12 18

### CODE REVIEW

File: app/views/administrate/application/_search.html.erb [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-    <% end %>
-  </span>
+  </label>
+
+  <input class="search__input"
+         id="search"
+         type="search"
+         name="search"
+         placeholder="Search <%= resource_name %>"
+         value="<%= search_term %>">
```
Comment (BenMorganIO) [inline app/views/administrate/application/_search.html.erb]: What would happen if I have a search term with a `"` in it?

File: app/views/administrate/application/_search.html.erb [hunk 2]
```diff
@@ -1,25 +1,22 @@
 <form class="search" role="search">
-  <span class="search__icon">
-    <svg class="search__clear-icon">
+  <label class="search__label" for="search">
+    <svg class="search__eyeglass-icon" role="img">
+      <title>Search <%= resource_name %></title>
```
Comment (krzysiek1507) [inline app/views/administrate/application/_search.html.erb]: Maybe I18n `Search %{query}`?

File: app/views/administrate/application/_search.html.erb [hunk 3]
```diff
@@ -1,25 +1,22 @@
...
-      </svg>
-    <% end %>
-  </span>
+  </label>
+
+  <input class="search__input"
+         id="search"
+         type="search"
+         name="search"
+         placeholder="Search <%= resource_name %>"
```
Comment (krzysiek1507) [inline app/views/administrate/application/_search.html.erb]: Maybe I18n `Search %{query}`?

File: app/views/administrate/application/_search.html.erb [hunk 4]
```diff
@@ -1,25 +1,22 @@
...
+         value="<%= search_term %>">
+
+  <%= link_to clear_search_params, class: "search__clear-link" do %>
+    <svg class="search__clear-icon" role="img">
+      <title>Clear search</title>
```
Comment (krzysiek1507) [inline app/views/administrate/application/_search.html.erb]: Maybe I18n?

File: app/views/administrate/application/_search.html.erb (hunk 1 above)
Comment (tysongach) [inline app/views/administrate/application/_search.html.erb]: It appears to work just fine…

FWIW, this PR doesn’t change this `value`, so if we haven’t had any bug reports, I think we’re good.

File: app/views/administrate/application/_search.html.erb (hunk 2 above)
Comment (tysongach) [inline app/views/administrate/application/_search.html.erb]: Oh, good call! Thanks.

File: app/views/administrate/application/_search.html.erb (hunk 1 above)
Comment (BenMorganIO) [inline app/views/administrate/application/_search.html.erb]: Sweet 👍

**Comment by tysongach** [comment]
This is ready for review again. I used Google Translate for the translations, if anyone has any improvements, I’m all ears…

**Comment by nickcharlton** [comment]
@tysongach I've just merged in a bunch of things which looks like it's just given you a bit of a rebase on this.

**Comment by tysongach** [comment]
@nickcharlton Rebased. This PR effectively overwrites the work in https://github.com/thoughtbot/administrate/pull/896 and https://github.com/thoughtbot/administrate/pull/891, if they are released in the same cycle.

**Pull request merged**